packages/adapter-components/src/client/http_connection.ts

```diff
@@ -69,20 +69,32 @@ export type ConnectionCreator<TCredentials> = (
   retryOptions: RetryOptions,
 ) => Connection<TCredentials>
 
+const getRetryDelayFromHeaders = (headers: Record<string, string>): number | undefined => {
+  if (headers['retry-after'] !== undefined) {
+    return parseInt(headers['retry-after'], 10) * 1000
+  } if (
```

Comment:
nit:
1. this is not an `else if`, so the `if` should start in its own line
```suggestion
  }
  if (
```
2. please add a comment explaining the use case and link to the docs, as this seems less standard